src/pocketmine/command/Command.php

```diff
@@ -239,19 +239,43 @@ public function getPermissionMessage() : ?string{
 	}
 
 	/**
+	 * Returns the generic description message of this command.
+	 *
 	 * @return string
 	 */
 	public function getDescription() : string{
 		return $this->description;
 	}
 
 	/**
+	 * Returns the specific description message of this command for the CommandSender at this moment.
+	 *
+	 * @param CommandSender $sender
+	 * @return string
+	 */
+	public function getDescriptionFor(CommandSender $sender) : string{
+		return $this->getDescription();
+	}
+
+	/**
+	 * Returns the generic usage message of this command.
+	 *
 	 * @return string
 	 */
 	public function getUsage() : string{
 		return $this->usageMessage;
 	}
 
+	/**
+	 * Returns the specific usage message of this command for the CommandSender at this moment.
+	 *
+	 * @param CommandSender $sender
+	 * @return string
+	 */
+	public function getUsageFor(CommandSender $sender) : string{
+		return $this->getUsage();
```

Comment:
Hardcoded usage messages will be removed at some point in the future. I'm not entirely happy with adding API that will inevitably get removed later on.